app/models/season.rb

```diff
@@ -0,0 +1,118 @@
+class Season < ApplicationRecord
+  has_many :games, dependent: :destroy
+
+  enum status: [:scheduled, :in_progress, :final]
+
+  validates :name, presence: true
+
+  after_initialize :set_default_values
+
+  def play!
+    puts "starting season..."
+    self.update(status: Season.statuses[:in_progress])
+    games.each { |g| g.play! }
+    self.update(status: Season.statuses[:final])
+    puts "season complete!"
+  end
+
+  def report!
+    print_standings
+    print_avg_leaders
+    %w(h rbi bb so doubles triples hr).each do |stat|
+      print_stat_leaders(stat)
+    end
+    nil
+  end
+
+  private
+
+  def set_default_values
+    self.status ||= Season.statuses[:scheduled]
+  end
+
+  def print_standings
```

Comment:
Clearly we need to change the data modeling to make tracking standings easier, but I'm okay punting on this for now.